File: machine/recommendation system/python/test_model.py
import numpy as np
from tensorflow.keras.models import load_model
import json
from de_contract import *


def bag_of_words(message):
    with open("words.json", encoding='utf-8') as file:
        words = json.load(file)
    message = replace(message)
    message_words = message.split()
    bag = [0] * len(words)

    for word in message_words:
        for i, w in enumerate(words):
            if word == w:
                bag[i] = 1

    return np.array(bag)


def predict_class(message):
    model = load_model("model.h5")
    with open("classes.json", encoding='utf-8') as file:
        classes = json.load(file)
    
    response = model.predict(np.array([bag_of_words(message)]))[0]
    results = [[i, r] for i, r in enumerate(response)]
    results.sort(key=lambda x: x[1], reverse=True)

    return [classes[r[0]] for r in results[:5]]

File: machine/recommendation system/node/test_model.js
const tf = require("@tensorflow/tfjs-node");
const { replace } = require("./de_contract");

const bag_of_words = (message) => {
  const words = require("./words.json");
  message = replace(message);
  message = message.split(" ");
  let bag = Array(words.length).fill(0);
  for (word of message) {
    if (words.includes(word)) {
      bag[words.indexOf(word)] = 1;
    }
  }
  return bag;
};

const predict_class = async (message) => {
  const classes = require("./classes.json");
  const model = await tf.loadLayersModel("file://model.json");
  const predict = model.predict(tf.tensor([bag_of_words(message)]));
  const result = await predict.dataSync();
  const array = Object.entries(result)
    .sort((a, b) => b[1] - a[1])
    .slice(0, 5);

  return array.map((item) => classes[item[0]]);
};

module.exports = predict_class;
